File: 2022-2023/sourcecode/lab/lab12/ValueInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/valueinjectbasicsjava/di/ComputerComponent.java
package pl.edu.uwr.pum.valueinjectbasicsjava.di;

import dagger.Component;
import pl.edu.uwr.pum.valueinjectbasicsjava.MainActivity;
import pl.edu.uwr.pum.valueinjectbasicsjava.di.modules.GpuModule;

@Component(modules = GpuModule.class)
public interface ComputerComponent {
    void inject(MainActivity mainActivity);
}

File: 2022-2023/sourcecode/lab/lab12/ValueInjectBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/valueinjectbasicskotlin/di/ComputerComponent.kt
package pl.edu.uwr.pum.valueinjectbasicskotlin.di

import dagger.Component
import pl.edu.uwr.pum.valueinjectbasicskotlin.MainActivity
import pl.edu.uwr.pum.valueinjectbasicskotlin.di.modules.GpuModule

@Component(modules = [GpuModule::class])
interface ComputerComponent {
    fun inject(mainActivity: MainActivity)
}